### PR TITLE
Fix/audit report

### PR DESCRIPTION
**Pull request by noyyyy**

### CODE REVIEW

File: src/RewardVault.sol [hunk 1]
```diff
@@ -23,12 +23,10 @@ contract RewardVault is IRewardVault, Ownable {
...
-        external
-        virtual
-        override
-        onlyOwner
-    {
+    function reward(
+        address to,
+        uint256 amount
+    ) external virtual override onlyOwner {
         IERC20(rebornToken).safeTransfer(to, amount);
```
Comment (yuanjunliang) [inline src/RewardVault.sol:30]: suggest emit event

File: src/interfaces/IRebornPortal.sol [hunk 2]
```diff
@@ -124,11 +126,17 @@ interface IRebornDefination {
 
     event Drop(uint256[] tokenIds);
 
+    /// @dev event about the vault address is set
+    event VaultSet(RewardVault);
```
Comment (yuanjunliang) [inline src/interfaces/IRebornPortal.sol:130]: event VaultSet(address rewardVault);

File: src/interfaces/IRebornPortal.sol [hunk 3]
```diff
@@ -1,6 +1,8 @@
 // SPDX-License-Identifier: GPL-3.0
 pragma solidity 0.8.17;
 
+import {RewardVault} from "src/RewardVault.sol";
```
Comment (yuanjunliang) [inline src/interfaces/IRebornPortal.sol:4]: `import {RewardVault} from "src/RewardVault.sol";`
delete

File: src/RewardVault.sol (hunk 1 above)
Comment (noyyyy) [reply src/RewardVault.sol:30]: Transfer itself emits an event. I think no extra events are needed

Comment (yuanjunliang) [reply src/RewardVault.sol:30]: It is helpful for notify user or background collection,just suggestion. if no need, ignore it

**Review by yuanjunliang — APPROVED** [review]

**Review by yuanjunliang — DISMISSED** [review]

**Review by yuanjunliang — APPROVED** [review]

**Pull request merged**